File: 268_Missing_Number.js
var missingNumber = function(nums) {
    let arr = [];
    nums = nums.sort((a,b) => a-b);
    let i = 0;
    let j = 0;
    while(j < nums.length){
        if(nums[j] !== i) arr.push(i);
        else j++;
        i++;
    }
    if(nums.length !== nums[nums.length - 1]) arr.push(nums.length);
    return arr;
};

File: 268_Missing_Number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int totalSum = (n * (n + 1)) / 2;
        int sum = 0;
        for(int i : nums){
            sum += i;
        }
        return totalSum - sum;
    }
};
